ImageGallery/Controllers/HomeController.cs

```diff
@@ -11,15 +11,20 @@ namespace ImageGallery.Controllers
 {
     public class HomeController : Controller
     {
-        private IStorageService _service;
-        public HomeController(IStorageService service)
+        private readonly IStorageService _storageService;
+        private readonly ICognitiveService _cognitiveService;
+
+        public HomeController(IStorageService storageService, ICognitiveService cognitiveService)
         {
-            _service = service;
+            _storageService = storageService;
+            _cognitiveService = cognitiveService;
         }
+
         public ActionResult Index()
         {
-            var imageCollection = _service.GetImages();
+            var imageCollection = _storageService.GetImages();
             ViewBag.ImageCollection = imageCollection;
+
             return View();
```

Comment:
Oops. I realized I should have passed the model to View as opposed to pass via the ViewBag since the former is strongly-typed. Could you fix that while you're at it?